hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetTable.java

```diff
@@ -122,28 +127,90 @@ protected void init() {
+        }
+    }
+
+    @EventBusListenerMethod(scope = EventScope.SESSION)
+    void onTargetUpdatedEvents(final TargetUpdatedEventContainer holder) {
+        onTargetUpdateEvents(holder.getEvents().stream().map(targetInfoUpdateEvent -> targetInfoUpdateEvent.getEntity())
+                .collect(Collectors.toList()));
+    }
+
+    @EventBusListenerMethod(scope = EventScope.SESSION)
+    void onTargetInfoUpdateEvents(final TargetInfoUpdateEventContainer holder) {
+        onTargetUpdateEvents(
+                holder.getEvents().stream().map(targetInfoUpdateEvent -> targetInfoUpdateEvent.getEntity().getTarget())
+                        .collect(Collectors.toList()));
+    }
+
     /**
-     * EventListener method which is called when a list of events is published.
-     * Event types should not be mixed up.
+     * EventListener method which is called by the event bus to notify about a
+     * list of {@link TargetInfoUpdateEvent}.
      *
-     * @param events
-     *            list of events
+     * @param updatedTargets
+     *            list of updated targets
      */
-    @EventBusListenerMethod(scope = EventScope.SESSION)
-    public void onEvents(final List<?> events) {
-        final Object firstEvent = events.get(0);
-        if (TargetCreatedEvent.class.isInstance(firstEvent)) {
-            onTargetCreatedEvents();
-        } else if (TargetInfoUpdateEvent.class.isInstance(firstEvent)) {
-            onTargetUpdateEvents(((List<TargetInfoUpdateEvent>) events).stream()
-                    .map(targetInfoUpdateEvent -> targetInfoUpdateEvent.getEntity().getTarget())
-                    .collect(Collectors.toList()));
-        } else if (TargetDeletedEvent.class.isInstance(firstEvent)) {
-            onTargetDeletedEvent((List<TargetDeletedEvent>) events);
-        } else if (TargetUpdatedEvent.class.isInstance(firstEvent)) {
-            onTargetUpdateEvents(((List<TargetUpdatedEvent>) events).stream()
-                    .map(targetInfoUpdateEvent -> targetInfoUpdateEvent.getEntity()).collect(Collectors.toList()));
+    private void onTargetUpdateEvents(final List<Target> updatedTargets) {
+        @SuppressWarnings("unchecked")
+        final List<Object> visibleItemIds = (List<Object>) getVisibleItemIds();
+
+        if (isFilterEnabled()) {
+            LOG.debug("Filter enabled on UI {}. Refresh targets from database.", getUI().getUIId());
+            refreshTargets();
+        } else {
+            updatedTargets.stream().filter(target -> visibleItemIds.contains(target.getTargetIdName()))
+                    .forEach(target -> updateVisibleItemOnEvent(target.getTargetInfo()));
         }
+
+        // workaround until push is available for action
+        // history, re-select
+        // the updated target so the action history gets
+        // refreshed.
+        final Optional<Target> selected = updatedTargets.stream()
```

Comment:
The code for the mentioned "workaround" is the same as already written in `onCancelTargetAssignmentEvents(...)`. So extract it to a private method and provide a proper name for the workaround as method name